nb_conda_kernels/tests/test_conda_kernels.py

```diff
@@ -14,14 +17,63 @@ def test_kernel_specs():
     python_envs = ["Python [{}]".format(env) for env in envs]
     r_envs = ["R [{}]".format(env) for env in envs]
 
+    return python_envs, r_envs
+
+
+def _notebook():
+    """Start the notebook server"""
     nb = subprocess.Popen(["jupyter", "notebook", "--no-browser", "--port=8999"])
     time.sleep(5)
 
-    kspecs_info = requests.get("http://localhost:8999/api/kernelspecs").json()
+    return nb
+
 
+def _get_kspecs():
+    """Ping the REST point to get kernelspecs info"""
+    kspecs_info = requests.get("http://localhost:8999/api/kernelspecs").json()
     kspecs = list(kspecs_info["kernelspecs"].keys())
 
+    return kspecs
+
+
+def _refresh():
+    driver = webdriver.Firefox()
+    driver.get("http://localhost:8999/api/kernelspecs")
+    driver.refresh()
+
+
+def test_kernel_specs():
+    """Test if the kernelspecs created on the fly belongs to the _setup set"""
+    python_envs, r_envs = _setup()
+    nb = _notebook()
+    kspecs = _get_kspecs()
+
     for kspec in kspecs:
         assert kspec in python_envs + r_envs
 
     nb.kill()
+
+
+def test_kernel_specs_update():
+    """Test if updating envs is detected by our manager creating the
+    corresponfing kernelspec"""
+    python_envs, r_envs = _setup()
+    nb = _notebook()
+    kspecs = _get_kspecs()
+
+    updated_py_env = "Python [{}]".format("_test_nb_conda_kernels")
+    assert updated_py_env not in kspecs
+
+    subprocess.Popen(["conda", "create", "-n", "_test_nb_conda_kernels", "ipykernel", "--yes"])
```

Comment:
Should we wait for the subprocess to complete here? Maybe use subprocess.check_call?